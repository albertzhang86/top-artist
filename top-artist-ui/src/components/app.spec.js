import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import App from './app.js';

describe('App', () => {
    it('should render correctly', () => {
        const component = create(
            <App>
                <div>Children of the App</div>
            </App>
        );
        expect(component).toMatchSnapshot();
    });

    it('should re-render when props change', () => {
        const wrapper = shallow(
            <App>
                <div>The original string</div>
            </App>
        );

        wrapper.setProps({
            children: <div>A new string</div>,
        });

        expect(wrapper.text()).toBe('A new string');
    });
});
