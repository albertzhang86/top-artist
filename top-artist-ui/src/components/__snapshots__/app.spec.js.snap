// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-offset-0 col-sm-offset-2 col-xs-12 col-sm-8"
  >
    <div>
      Children of the App
    </div>
  </div>
</div>
`;
